tests/mockserver.py

```diff
@@ -68,7 +68,7 @@ def get_received(self):
         self._buf.seek(0)
         # TODO: have to process string encoding properly. currently we assume
         # that all encoding is utf-8.
-        return list(Unpacker(self._buf, encoding='utf-8'))
+        return list(Unpacker(self._buf))
```

Comment:
Changing the mock server to "make tests pass" does nothing to alleviate concerns in #157